addons/account/models/account_account.py

```diff
@@ -1040,194 +1041,281 @@ def get_import_templates(self):
+        if forbidden_companies := (self.sudo().company_ids - self.env.companies):
             raise UserError(_(
                 "You do not have the right to perform this operation as you do not have access to the following companies: %s.",
                 ", ".join(c.name for c in forbidden_companies)
             ))
 
-        # Step 3: Update records in DB.
-        accounts_to_remove = self - account_to_merge_into
+        # Step 2: Create new accounts.
+        base_company = self.env.company if self.env.company in self.company_ids else self.company_ids[0]
+        companies_to_update = self.company_ids - base_company
+        check_company_fields = {fname for fname, field in self._fields.items() if field.relational and field.check_company}
+        new_account_by_company = {
+            company: self.copy(default={
+                'name': self.name,
+                'company_ids': [Command.set(company.ids)],
+                **{
+                    fname: self[fname].filtered(lambda record: record.company_id == company)
+                    for fname in check_company_fields
+                }
+            })
+            for company in companies_to_update
+        }
+        new_accounts = self.env['account.account'].union(*new_account_by_company.values())
 
-        # 3.1: Update foreign keys in DB
-        wiz = self.env['base.partner.merge.automatic.wizard'].new()
-        wiz._update_foreign_keys_generic('account.account', accounts_to_remove, account_to_merge_into)
+        # Step 3: Update foreign keys in DB.
 
-        # 3.2: Update Reference and Many2OneReference fields that reference account.account
-        wiz._update_reference_fields_generic('account.account', accounts_to_remove, account_to_merge_into)
+        new_account_id_by_company_id = {str(company.id): new_account.id for company, new_account in new_account_by_company.items()}
+        new_account_id_by_company_id[base_company.id] = self.id  # Needed for update of xmlids
+        new_account_id_by_company_id_json = json.dumps(new_account_id_by_company_id)
+        (self | new_accounts).invalidate_recordset()
 
-        # Step 4: Remove merged accounts
-        self.env.invalidate_all()
+        # 3.1: Add fields on other models that reference account.account
+        many2x_fields = self.env['ir.model.fields'].search([
+            ('ttype', 'in', ('many2one', 'many2many')),
+            ('relation', '=', 'account.account'),
+            ('store', '=', True)
+        ])
+        for field_to_update in many2x_fields:
+            model = field_to_update.model
+            if not self.env[model]._auto:
+                continue
+            if model == 'res.company':
+                company_id_field = 'id'
+            elif 'company_id' in self.env[model]:
+                company_id_field = 'company_id'
+            else:
+                continue
+            if not self.env[model]._fields[company_id_field].store:
+                continue
```

Comment:
This could possibly lead to inconsistencies but I don't have a solution

Reply:
I changed it a bit so that it uses `_field_to_sql` on the `company_id_field`, so if this causes any issues, we can just implement `_field_to_sql` where it is needed.